node/pod.go

```diff
@@ -276,3 +261,37 @@ func (pc *PodController) podStatusHandler(ctx context.Context, key string) (retE
 
 	return pc.updatePodStatus(ctx, pod, key)
 }
+
+func (pc *PodController) deletionReconcilationWorkItemHandler(ctx context.Context, key string) error {
+	ctx, span := trace.StartSpan(ctx, "processDeletionReconcilationWorkItem")
+	defer span.End()
+	// Add the current key as an attribute to the current span.
+	ctx = span.WithField(ctx, "key", key)
+
+	// Convert the namespace/name string into a distinct namespace and name.
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		// Log the error as a warning, but do not requeue the key as it is invalid.
+		log.G(ctx).Warn(pkgerrors.Wrapf(err, "invalid resource key: %q", key))
+		return nil
+	}
+
+	_, err = pc.podsLister.Pods(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+
+	// We check if the pod still exists in the provider. If it no longer there, then we ensure that it's
+	// not in the API server.
+	pod, err := pc.provider.GetPod(ctx, namespace, name)
+	if err != nil && !errdefs.IsNotFound(err) {
+		return err
+	}
+	if err == nil && pod != nil {
+		log.G(ctx).Warn("Container still exists in provider, requeueing")
+		pc.deletionQ.AddRateLimited(key)
+		return nil
+	}
+
+	return pc.client.Pods(namespace).Delete(name, metav1.NewDeleteOptions(0))
```

Comment:
Error checking logic above does not look right.
If there is any error then we'll delete the pod but I think we should only be doing this if the error is "not found".